packages/tupaia-web/src/api/mutations/useEmailDashboard.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+import { useMutation } from 'react-query';
+import { API_URL, post } from '../api';
+import { Dashboard, DashboardItem, EntityCode, ProjectCode } from '../../types';
+
+type EmailDashboardParams = {
+  projectCode?: ProjectCode;
+  entityCode?: EntityCode;
+  dashboardName?: Dashboard['name'];
+  dashboardId?: Dashboard['id'];
```

Comment:
We have to send through the dashboardId separately, since we use the `dashboardName` in the tupaia urls. I took a quick looking to refactoring things so that we use the unique `dashboard.code` in the url instead, but it looked a little awkward for when there's a multi-country project with a default dashboard. So left it for now.